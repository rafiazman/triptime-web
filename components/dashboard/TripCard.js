/** @format */

import React from 'react';
import styles from '../../css/trip-card.module.css';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircle } from '@fortawesome/free-solid-svg-icons';
import Tooltip from '../Tooltip';

export default class TripCard extends React.Component {
  render() {
    return (
      <div className={styles.card}>
        <h1>{this.props.tripName}</h1>
        <div className={styles.alerts}>
          {this.generateAlert(this.props.updated)}
        </div>
      </div>
    );
  }

  generateAlert(updated) {
    if (!updated) return '';
    return (
      <Tooltip
        text={'Trip updated since you last checked'}
        component={
          <span>
            <FontAwesomeIcon icon={faCircle} />
          </span>
        }
      />
    );
  }
}

TripCard.propTypes = {
  tripName: PropTypes.string.isRequired,
  updated: PropTypes.bool.isRequired,
};
