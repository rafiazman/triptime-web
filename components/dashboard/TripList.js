/** @format */

import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import TripCard from './TripCard';
import styles from '../../css/homepage.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ReactLoading from 'react-loading';

export default class TripList extends React.Component {
  render() {
    const tripInfoList = this.props.tripInfoList;
    return (
      <div className={styles.component}>
        <h2>
          {' '}
          <FontAwesomeIcon icon={this.props.icon} /> {this.props.title}{' '}
        </h2>
        {this.props.loading ? (
          <div className={styles.loadingContainer}>
            <ReactLoading type='bubbles' color='#ff4200' />
          </div>
        ) : tripInfoList.length > 0 ? (
          <div>
            {tripInfoList.map((tripInfo, index) => (
              <Link href={`/trip/${tripInfo.id}`} key={index}>
                <a>
                  <TripCard
                    tripName={tripInfo.name}
                    updated={tripInfo.updated}
                  />
                </a>
              </Link>
            ))}
          </div>
        ) : (
          this.props.displayIfNoTrip
        )}
      </div>
    );
  }
}

TripList.propTypes = {
  displayIfNoTrip: PropTypes.element.isRequired,
  icon: PropTypes.object.isRequired,
  tripInfoList: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
};
