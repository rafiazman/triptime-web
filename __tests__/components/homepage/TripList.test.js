/** @format */

import TestRenderer from 'react-test-renderer';
import React from 'react';
import TripList from '../../../components/dashboard/TripList';
import TripCard from '../../../components/dashboard/TripCard';
import myTripInfos from '../../../app/dummy-data/my-trip-infos';
import { faCode } from '@fortawesome/free-solid-svg-icons';

const testNoTripDisplay = (
  <h3 className={'no-display'}>
    This should be displayed when there is no trip
  </h3>
);

describe('Test TripList renders correctly when given no trip', () => {
  const noTripRenderer = TestRenderer.create(
    <TripList
      displayIfNoTrip={testNoTripDisplay}
      icon={faCode}
      tripInfoList={[]}
      title={'Test Trip List'}
      loading={false}
    />,
  );
  const noTripInstance = noTripRenderer.root;

  test('TripList renders no TripCard if given no trip', () => {
    expect(noTripInstance.findAllByType(TripCard).length).toBe(0);
  });
  test('TripList renders a message if given no trip', () => {
    expect(
      noTripInstance.findByProps({ className: 'no-display' }).innerHTML,
    ).toEqual(testNoTripDisplay.innerHTML);
  });
  test('Match with Snapshot', () => {
    expect(noTripRenderer.toJSON()).toMatchSnapshot();
  });
});

describe('Test TripList renders correctly when given four trips', () => {
  const fourTripsRenderer = TestRenderer.create(
    <TripList
      displayIfNoTrip={testNoTripDisplay}
      icon={faCode}
      tripInfoList={myTripInfos}
      title={'Test Trip List'}
      loading={false}
    />,
  );
  const fourTripsInstance = fourTripsRenderer.root;

  test('TripList renders four TripCard if given four trips', () => {
    expect(fourTripsInstance.findAllByType(TripCard).length).toBe(4);
  });
  test('TripList does not render a noTripMessage if given four trips', () => {
    expect(
      fourTripsInstance.findAllByProps({ className: 'no-display' }).length,
    ).toBe(0);
  });
  test('Match with Snapshot', () => {
    expect(fourTripsRenderer.toJSON()).toMatchSnapshot();
  });
});
