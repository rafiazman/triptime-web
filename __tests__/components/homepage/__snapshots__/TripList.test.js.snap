// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TripList renders correctly when given four trips Match with Snapshot 1`] = `
<div
  className="component"
>
  <h2>
     
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-code fa-w-20 "
      data-icon="code"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     
    Test Trip List
     
  </h2>
  <div>
    <a
      href="/trip/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="card"
      >
        <h1>
          test1
        </h1>
        <div
          className="alerts"
        >
          <span
            className="tooltip"
          >
            <span>
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle fa-w-16 "
                data-icon="circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <div
              className="text"
            >
              Trip updated since you last checked
            </div>
          </span>
        </div>
      </div>
    </a>
    <a
      href="/trip/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="card"
      >
        <h1>
          test2
        </h1>
        <div
          className="alerts"
        >
          
        </div>
      </div>
    </a>
    <a
      href="/trip/3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="card"
      >
        <h1>
          test3
        </h1>
        <div
          className="alerts"
        >
          <span
            className="tooltip"
          >
            <span>
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle fa-w-16 "
                data-icon="circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <div
              className="text"
            >
              Trip updated since you last checked
            </div>
          </span>
        </div>
      </div>
    </a>
    <a
      href="/trip/4"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="card"
      >
        <h1>
          test4
        </h1>
        <div
          className="alerts"
        >
          
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Test TripList renders correctly when given no trip Match with Snapshot 1`] = `
<div
  className="component"
>
  <h2>
     
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-code fa-w-20 "
      data-icon="code"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     
    Test Trip List
     
  </h2>
  <h3
    className="no-display"
  >
    This should be displayed when there is no trip
  </h3>
</div>
`;
